src/world/sprite.c

```diff
@@ -39,7 +52,45 @@ void create_balloon(int x, int y, int z, int colour)
  */
 void create_duck(int targetX, int targetY)
 {
-	RCT2_CALLPROC_X(0x0067440F, targetX, 0, targetY, 0, 0, 0, 0);
+	rct_sprite* sprite = create_sprite(2);
+	if (sprite != NULL)
+	{
+		sprite->duck.sprite_identifier = 2;
+		sprite->duck.var_01 = 8;
+		sprite->duck.var_14 = 9;
+		sprite->duck.var_09 = 0xC;
+		sprite->duck.var_15 = 9;
+		int ebx = scenario_rand();
+		int ebp = ebx;
+		int edi = ebx;
+		ebp >>= 8;
+		edi >>= 16;
+		edi &= 0x3F;
+		ebp &= 0x1E;
+		targetX += ebp;
+		targetY += ebp;
+		sprite->duck.x = targetX;
+		sprite->duck.y = targetY;
+		switch (ebx & 3)
+		{
+		case 0:
+			targetX = 8191 - edi;
+			break;
+		case 1:
+			targetY = edi;
+			break;
+		case 2:
+			targetX = edi;
+			break;
+		case 3:
+			targetY = 8191 - edi;
+			break;
+		}
+		sprite->duck.var_1E = (ebx << 3) & 0xFF;
```

Comment:
Small bug here. ebx has been &'d with 3 before this function for the above switch statement.
Suggest changing the switch to:

```
uint8 direction = scenario_rand() & 3;
switch(direction){
...
}
var_1E = direction << 3;